ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/message/CreateFunctionHandler.java

```diff
@@ -127,8 +128,11 @@ private CreateFunctionDesc build() {
       String fullQualifiedFunctionName = FunctionUtils.qualifyFunctionName(
           metadata.function.getFunctionName(), destinationDbName
       );
+      // For bootstrap load, the create function should be always performed.
+      // Only for incremental load, need to validate if event is newer than the database.
+      ReplicationSpec replSpec = (context.dmd == null) ? null : eventOnlyReplicationSpec(context);
```

Comment:
I think for bootstrap load case, dmd will be null and I saw similar check in the caller which was added by you. :)
Also, in case of bootstrap, we shouldn't check for the database object's lastReplID.